src/js/tracks/text-track-display.js

```diff
@@ -32,6 +32,7 @@ class TextTrackDisplay extends Component {
     super(player, options, ready);
 
     player.on('loadstart', Lib.bind(this, this.toggleDisplay));
+    player.on('texttrackchange', Lib.bind(this, this.toggleDisplay));
```

Comment:
Ha, ignore my previous comment about the TODO. We can remove the TODO line with this change and it will close the related issue.